File: dayinshalou.py
from math import sqrt

'''
打印沙漏
'''
def print_img(num, ch):
    # up
    for i in range(num, 0, -1):
        for j in range(num - i, 0, -1):
            print(" ", end="")
        for j in range(2 * i - 1):
            print(ch, end="")
        print()
    # down
    for i in range(2, num + 1):
        for j in range(num - i, 0, -1):
            print(" ", end="")
        for j in range(2 * i - 1):
            print(ch, end="")
        print()


if __name__ == "__main__":
    s = input()
    a, b = s.split(" ")
    ceng = int(sqrt((int(a) + 1) / 2))
    print_img(ceng, b)
    print(int(a) - 2 * ceng * ceng + 1)

File: dayinshalou.c
#include <stdio.h>
#include <math.h>

// gcc dayinshalou.c -o  dayinshalou -lm
void print_img(int num, char ch) {
    int i, j;
    //up
    for (i = num; i >= 1; i--) {
        for (j = num - i; j > 0; j--)
            printf(" ");
        for (j = 0; j < 2 * i - 1; j++)
            printf("%c", ch);
        printf("\n");
    }
    //down
    for (i = 2; i <= num; i++) {
        for (j = num - i; j > 0; j--)
            printf(" ");
        for (j = 0; j < 2 * i - 1; j++)
            printf("%c", ch);
        printf("\n");
    }
}


int main() {
    int num, ceng;
    char ch;
    scanf("%d %c", &num, &ch);
    ceng = (int) sqrt((num + 1) / 2);
    print_img(ceng, ch);
    printf("%d\n", num - 2 * ceng * ceng + 1);
    return 0;
}